rcl/test/rcl/allocator_testing_utils.h

```diff
@@ -87,6 +87,80 @@ set_failing_allocator_is_failing(rcutils_allocator_t & failing_allocator, bool s
   ((__failing_allocator_state *)failing_allocator.state)->is_failing = state;
 }
 
+typedef struct __time_bomb_allocator_state
+{
+  int count_until_failure;
+} __time_bomb_allocator_state;
+
+void * time_bomb_malloc(size_t size, void * state)
+{
+  __time_bomb_allocator_state * time_bomb_state = (__time_bomb_allocator_state *)state;
+  if (time_bomb_state->count_until_failure >= 0 &&
+    time_bomb_state->count_until_failure-- == 0)
+  {
+    return nullptr;
+  }
+  return rcutils_get_default_allocator().allocate(size, rcutils_get_default_allocator().state);
+}
+
+void *
+time_bomb_realloc(void * pointer, size_t size, void * state)
+{
+  __time_bomb_allocator_state * time_bomb_state = (__time_bomb_allocator_state *)state;
+  if (time_bomb_state->count_until_failure >= 0 &&
+    time_bomb_state->count_until_failure-- == 0)
+  {
+    return nullptr;
+  }
+  return rcutils_get_default_allocator().reallocate(
+    pointer, size, rcutils_get_default_allocator().state);
+}
+
+void
+time_bomb_free(void * pointer, void * state)
+{
+  __time_bomb_allocator_state * time_bomb_state = (__time_bomb_allocator_state *)state;
+  if (time_bomb_state->count_until_failure >= 0 &&
+    time_bomb_state->count_until_failure-- == 0)
+  {
+    return;
+  }
+  rcutils_get_default_allocator().deallocate(pointer, rcutils_get_default_allocator().state);
+}
+
+void *
+time_bomb_calloc(size_t number_of_elements, size_t size_of_element, void * state)
+{
+  __time_bomb_allocator_state * time_bomb_state = (__time_bomb_allocator_state *)state;
+  if (time_bomb_state->count_until_failure >= 0 &&
+    time_bomb_state->count_until_failure-- == 0)
+  {
+    return nullptr;
+  }
+  return rcutils_get_default_allocator().zero_allocate(
+    number_of_elements, size_of_element, rcutils_get_default_allocator().state);
+}
+
+static inline rcutils_allocator_t
+get_time_bombed_allocator(void)
+{
+  static __time_bomb_allocator_state state;
```

Comment:
why static?

Reply:
See d1c4cb0

Reply:
Actually I put it back with a034d74. This is needed to keep the variable state that's modified with the other functions of the file.